server/sonar-server/src/main/java/org/sonar/server/user/UserSession.java

```diff
@@ -127,6 +138,12 @@
   UserSession checkComponentUuidPermission(String permission, String componentUuid);
 
   /**
+   * Whether the user has the permission on the component. Returns {@code false}
+   * if the component does not exist in database.
```

Comment:
"Returns {@code false} whether the component does not exist or user does not have the specified permission on it."